### DISCUSSION TITLE
Converters should broadly accept string inputs that parse to the expected type

### DISCUSSION THREAD

**Discussion by BioTurboNick** [CommunityToolkit/Maui#765] [New Feature Discussions]
The converters I use for my own code all accept strings from `ConverterParameter`, because that's what e.g. `ConverterParameter=2` passes to the converter. It's very convenient. Would be nice if these converters tolerated and attempted to parse string input to match the bound data type, if such an operation is possible.

**Comment by TheCodeTraveler** [comment]
Hi @BioTurboNick! Can you link to an example of some code?

**Comment by BioTurboNick** [comment]
I only needed it for ints but here's an example:
```
public class AreEqualConverter :
    IValueConverter
{
    public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
    {
        if (targetType != typeof(bool))
            return new object();
        if (!(value is int intValue))
            return false;

        if (!(parameter is int otherValue))
        {
            bool parseSuccessful = int.TryParse(parameter as string, out otherValue);

            if (!parseSuccessful)
                return false;
        }

        return intValue == otherValue;
    }

    public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
    {
        throw new NotSupportedException();
    }
}
```

Another example

**Reply by bijington** [reply]
You can use integers in your XAML to prevent needing this requirement. It isn't the nicest to read but you can see an example of how we do it with a `Double` in this page:

https://github.com/CommunityToolkit/Maui/blob/4b4780357a2e7cb340a41312535553a12c160149/samples/CommunityToolkit.Maui.Sample/Pages/Converters/CompareConverterPage.xaml#L12

**Comment by CliffAgius** [comment]
Closed and answered
